obstacle_stop_planner_refine/include/obstacle_stop_planner/adaptive_cruise_control.hpp

```diff
@@ -172,22 +176,23 @@ class AdaptiveCruiseController
   };
   Param param_;
 
-  double getMedianVel(const std::vector<geometry_msgs::msg::TwistStamped> vel_que);
+  double getMedianVel(const std::vector<geometry_msgs::msg::TwistStamped> & vel_que);
   double lowpass_filter(const double current_value, const double prev_value, const double gain);
-  void calcDistanceToNearestPointOnPath(
+  double calcDistanceToNearestPointOnPath(
     const autoware_planning_msgs::msg::Trajectory & trajectory, const int nearest_point_idx,
-    const geometry_msgs::msg::Pose & self_pose, const pcl::PointXYZ & nearest_collision_point,
-    const rclcpp::Time & nearest_collision_point_time, double * distance);
+    const geometry_msgs::msg::Pose & self_pose, const Point2d & nearest_collision_point,
+    const rclcpp::Time & nearest_collision_point_time);
   double calcTrajYaw(
     const autoware_planning_msgs::msg::Trajectory & trajectory, const int collision_point_idx);
-  bool estimatePointVelocityFromObject(
+  std::tuple<bool, double> estimatePointVelocityFromObject(
```

Comment:
Probably can be replaced by `optional`?

Reply:
ROS foxy uses C++14, and `optional` is not included in it. Should I use another implementation like this? https://github.com/TartanLlama/optional

Reply:
Please use `boost::optional` and replace it by `std::optional` in ROS2 Galactic(which supports C++17)!

Reply:
I replaced tuple with optional. 030d10e
I think it's appropriate to use tuples in some parts of the code, so I've left them unchanged.